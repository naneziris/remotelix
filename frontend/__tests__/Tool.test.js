import ToolComponent from '../components/Tool';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeTool = {
  id: 'ABC123',
  title: 'A Cool tool',
  url: 'www.remotelix.com',
  description: 'This tool is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Tool/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ToolComponent tool={fakeTool} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<ToolComponent tool={fakeTool} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeTool.image);
    expect(img.props().alt).toBe(fakeTool.title);
  });

  it('renders the  title', () => {
    const wrapper = shallow(<ToolComponent tool={fakeTool} />);
    expect(wrapper.find('Title a').text()).toBe(fakeTool.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ToolComponent tool={fakeTool} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(2);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('DeleteTool').exists()).toBe(true);
  });
});
