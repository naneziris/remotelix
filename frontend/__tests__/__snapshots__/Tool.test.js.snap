// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tool/> renders and matches the snapshot 1`] = `
<ToolStyles__Tool>
  <img
    alt="A Cool tool"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/tool",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool tool
      </a>
    </Link>
  </Title>
  <p>
    This tool is really cool!
  </p>
  <p>
    www.remotelix.com
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <DeleteTool
      id="ABC123"
    >
      Delete This Tool
    </DeleteTool>
  </div>
</ToolStyles__Tool>
`;
